test/utils.sh

```diff
@@ -507,32 +531,61 @@ function verify_sockshop_deployment() {
   KEPTN_NAMESPACE=$5
   BLUE_GREEN_DEPLOYMENT=$6
 
-  echo "---------------------------------------------"
-  echo "Checking ${STAGE} deployment"
+  echo ""
+  echo "---------------------------------------------------------------------------------------------------------"
+  echo "Checking ${STAGE} deployment (namespace: ${PROJECT}-${STAGE}) for ${ARTIFACT_IMAGE}:${ARTIFACT_IMAGE_TAG}"
+  echo "---------------------------------------------------------------------------------------------------------"
   echo ""
 
+  echo "Pre-req: Checking if carts-db is already running..."
+
+  # verify that a carts-db deployment exists
   wait_for_deployment_in_namespace "carts-db" "${PROJECT}-${STAGE}"
   verify_test_step $? "Deployment carts-db not up in ${PROJECT}-${STAGE}, exiting ..."
-  wait_for_deployment_with_image_in_namespace "carts" "${PROJECT}-${STAGE}" "${ARTIFACT_IMAGE}:${ARTIFACT_IMAGE_TAG}"
-  verify_pod_in_namespace "carts" "${PROJECT}-${STAGE}"
-  verify_test_step $? "Pod carts not found, exiting ..."
+
+  # verify that a cards-db pod is up and running
+  verify_pod_in_namespace "carts-db" "${PROJECT}-${STAGE}"
+  verify_test_step $? "Pod carts-db not found in $${PROJECT}-${STAGE}, exiting ..."
+
+  echo ""
+  echo "Checking if carts is up and running..."
 
   if [[ "${BLUE_GREEN_DEPLOYMENT}" == "true" ]]; then
+    # verify that a carts deployment exists (but it does not need to be up, as we need to look for carts-primary)
+    wait_and_check_if_deployment_exists "carts" "${PROJECT}-${STAGE}"
+    verify_test_step $? "Deployment ${ARTIFACT_IMAGE}:${ARTIFACT_IMAGE_TAG} carts not found in ${PROJECT}-${STAGE}, exiting ..."
+
+    # verify that a blue-green carts deployment is up and running
     wait_for_deployment_with_image_in_namespace "carts-primary" "${PROJECT}-${STAGE}" "${ARTIFACT_IMAGE}:${ARTIFACT_IMAGE_TAG}"
+    verify_test_step $? "Deployment carts ${ARTIFACT_IMAGE}:${ARTIFACT_IMAGE_TAG} not up in ${PROJECT}-${STAGE}, exiting ..."
+
+    # verify that a blue-green carts pod is up and running
     verify_pod_in_namespace "carts-primary" "${PROJECT}-${STAGE}"
     verify_test_step $? "Pod carts-primary not found, exiting ..."
+
+    # verify image name of carts-primary deployment
+    verify_image_of_deployment "carts-primary" "${PROJECT}-${STAGE}" "${ARTIFACT_IMAGE}:$ARTIFACT_IMAGE_TAG"
```

Comment:
do we still need this check if we have verified the deployment/image with `wait_for_deployment_with_image_in_namespace` above?

Reply:
I double checked, you are correct, we don't need this here.